modules/core/src/main/java/org/apache/synapse/ExtendedSynapseHandler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.synapse;
+
+/**
+ * Interface For Extended Synapse Handlers
+ * <p/>
+ * Synapse Handlers are invoked when a message received to the mediation engine or a message sent out
+ * from the engine.
+ * <p/>
+ * When a message received to the engine, handles are invoked just before the mediation flow
+ * When a message is sent out from the engine, handlers are invoked just after the mediation flow
+ * There are two requests coming into the engine and two responses going out from the engine.
+ * For all four messages relevant method is invoked
+ */
+public interface ExtendedSynapseHandler extends SynapseHandler {
+
+    /**
+     * Handle error requests coming to the engine.
+     *
+     * @param synCtx outgoing response message context
+     * @return whether mediation flow should continue
+     */
+    public boolean handleError(MessageContext synCtx);
+
+    /**
+     * Handle server startup.
+     */
+    public boolean handleInit();
+
+    /**
+     * Handle server shut down.
+     */
+    public boolean handleStopServer();
```

Comment:
can both of these method adhere to same convention. one is with server and other without. Also please update the method comment accordingly. If this something related to deployment, shall we include that.

Reply:
Fixed